internal/controlplane/handlers_githubwebhooks.go

```diff
@@ -1547,25 +1562,38 @@ func (s *Server) reconcilePrWithDb(
 		})
 		if err != nil {
 			return nil, fmt.Errorf(
-				"cannot upsert PR %d in repo %s",
-				prEvalInfo.Number, repoEnt.Properties.GetProperty(properties.PropertyName).GetString())
+				"cannot upsert PR %d in repo %s: %w",
+				pullProps.GetProperty(ghprop.PullPropertyNumber).GetInt64(),
+				repoEnt.Properties.GetProperty(properties.PropertyName).GetString(),
+				err)
 		}
 		retErr = nil
 	case webhookActionEventClosed:
 		_, err := db.WithTransaction(s.store, func(t db.ExtendQuerier) (*db.PullRequest, error) {
 			err := t.DeletePullRequest(ctx, db.DeletePullRequestParams{
 				RepositoryID: repoEnt.Entity.ID,
-				PrNumber:     prEvalInfo.Number,
+				PrNumber:     pullProps.GetProperty(ghprop.PullPropertyNumber).GetInt64(),
 			})
 			if err != nil {
 				return nil, err
 			}
 
 			err = t.DeleteEntityByName(ctx, db.DeleteEntityByNameParams{
-				Name:      pullRequestName(repoEnt.Properties.GetProperty(properties.PropertyName).GetString(), prEvalInfo.Number),
+				Name:      pullName,
 				ProjectID: repoEnt.Entity.ProjectID,
 			})
-			if err != nil {
+			if err != nil && !errors.Is(err, sql.ErrNoRows) {
+				return nil, err
+			}
+
+			// temporary migration code to get rid of PRs with the old name
+			err = t.DeleteEntityByName(ctx, db.DeleteEntityByNameParams{
```

Comment:
Note that this PR /changes the name of the PRs/. We don't have to do that in this PR if we don't want to, but I since the old code just used repoName/pullNumber I wanted to address the case of two repos with the same name coming from different providers and yielding the same entity name.